usr/share/rear/init/default/03_update_recovery_system.sh

```diff
@@ -0,0 +1,42 @@
+
+# Updating the currently running system.
+
+# See https://github.com/rear/rear/issues/841
+
+# Currently updating rear is only supported during "rear recover"
+# because "rear recover" is only run once in the recovery system and
+# afterwards the recovery system is shut down and the recovered system is booted.
+# Currently it is not tested what mess might happen for other workflows
+# that run many times in one same system like "rear mkbackup" in the normal system.
+# Furthermore it seems to make not much sense to update rear in the normal system
+# via this special built-in rear functionality because in the normal system
+# one can manually update rear as anything else.
+test "$WORKFLOW" != "recover" && return
+
+# Without a RECOVERY_UPDATE_URL there is nothing to do:
+test "$RECOVERY_UPDATE_URL" || return
+
+# The actual work:
+
+# Tell the user that the recovery system will be updated:
+LogPrint "Updating recovery system with the content from '$RECOVERY_UPDATE_URL':"
+
+# Download the tar.gz that contains the update files.
+local update_archive_filename="recovery-update.tar.gz"
+# "curl -f" does not fail reliable (there are occasions where non-successful response codes will slip through)
+# therefore the HTTP response code is written to stdout ('-w') so that it can be explicitly tested afterwards
+# ('-s' does not show progress meter or error messages but 'S' it makes it show an error message if it fails)
+# "curl --verbose" messages go to stderr so that they go to the "rear recover" log file:
+local http_response_code=$( curl $verbose -f -s -S -w "%{http_code}" -o /$update_archive_filename $RECOVERY_UPDATE_URL )
```

Comment:
@jsmeix just make sure that `curl` is added to the `REQUIRED_PROGS` array if we depend on it in recovery mode.